instrumentation/scala-2.12.0/src/main/java/scala/concurrent/impl/CallbackRunnable_Instrumentation.java

```diff
@@ -25,15 +29,31 @@
 @Weave(originalName = "scala.concurrent.impl.CallbackRunnable")
 public class CallbackRunnable_Instrumentation<T> {
     private final Function1<Try<?>, Object> onComplete = Weaver.callOriginal();
-
     private Try<T> value = Weaver.callOriginal();
+    @NewField
+    private Token token;
+
+    public CallbackRunnable_Instrumentation(final ExecutionContext executor, final Function1 onComplete) {
+        Transaction transaction = AgentBridge.getAgent().getTransaction(false);
+        if (AgentBridge.activeToken.get() == null &&
+                transaction != null &&
+                NewRelic.getAgent().getConfig().getValue("scala.callbackrunnable.enabled", false) &&
+                AgentBridge.getAgent().getTracedMethod().trackCallbackRunnable()) {
+            token = transaction.getToken();
+        }
+    }
 
     /**
      * Override the setter for "value" so we can replace it with our wrapped version
      */
+
+    @Trace(async = true, excludeFromTransactionTrace = true)
     public void value_$eq(Try<T> original) {
         Weaver.callOriginal();
-
+        if (token != null) {
+            token.linkAndExpire();
```

Comment:
Is it safe for multiple threads to link and expire at the same time?  If not, there might be a race condition here and we may need to introduce a semaphore or other concurrent safety net.

Reply:
Callbackrunnables can only be executed once, so this shouldn't be an issue